pkg/api/addtask.go

```diff
@@ -0,0 +1,205 @@
+package api
+
+import (
+	"database/sql"
+	"encoding/json"
+	"final/pkg/db"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+func writeJson(w http.ResponseWriter, data any) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	json.NewEncoder(w).Encode(data)
+}
+
+func checkDate(task *db.Task) error {
+	now := time.Now()
+	today := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location())
+
+	if task.Date == "" {
+		task.Date = today.Format("20060102")
+		return nil
+	}
+
+	t, err := time.Parse("20060102", task.Date)
+	if err != nil {
+		return err
+	}
+
+	next, err := NextDate(today, task.Date, task.Repeat)
+	if err != nil {
+		return err
```

Comment:
Ошибку лучше всегда оборачивать, так быстрее будет найти, где она возникла.
```suggestion
		return fmt.Errorf("wrong repeat format: %w", err)
```